File: 05-02-binding-adapter/java-05-02-binding-adapter/app/src/main/java/com/example/bindingadapter/TextViewBindingAdapters.java
package com.example.bindingadapter;

import android.widget.TextView;

import androidx.databinding.BindingAdapter;

public class TextViewBindingAdapters {
    @BindingAdapter("reverseText")
    public static void setReverseText(TextView view, String toReverse) {
        String reversed = new StringBuilder(toReverse).reverse().toString();
        view.setText(reversed);
    }
}

File: 05-02-binding-adapter/kotlin-05-02-binding-adapter/app/src/main/java/com/example/bindingadapter/TextViewBindingAdapters.kt
package com.example.bindingadapter

import androidx.databinding.BindingAdapter
import android.widget.TextView

object TextViewBindingAdapters {
    @BindingAdapter("reverseText")
    fun setReverseText(view: TextView, toReverse: String) {
        val reversed = StringBuilder(toReverse).reverse().toString()
        view.text = reversed
    }
}
